internal/brokers/examplebroker/broker.go

```diff
@@ -432,9 +454,15 @@ func (b *Broker) IsAuthenticated(ctx context.Context, sessionID, authenticationD
 	}
 
 	// Store last successful authentication mode for this user in the broker.
-	b.userLastSelectedModeMu.Lock()
-	b.userLastSelectedMode[sessionInfo.username] = sessionInfo.selectedMode
-	b.userLastSelectedModeMu.Unlock()
+	if access == responses.AuthGranted {
+		b.userLastSelectedModeMu.Lock()
+		b.userLastSelectedMode[sessionInfo.username] = sessionInfo.selectedMode
```

Comment:
do we rzeally need that line? I mean, in case MFA == 1, it was still assign to firstSelectedMode, no?

Reply:
Yes and I thought a lot about whether we needed it or not, but in the end, I opted for not relying on that behavior here (to avoid failing if it changed in the future).

Reply:
Right, but I think the path of least complexity, telling "we always assign firstSelectedMode" is the most comphrensive way.